common/src/main/java/org/apache/nemo/common/ir/IRDAGChecker.java

```diff
@@ -348,8 +349,12 @@ void addStreamVertexCheckers() {
   }
 
   void addLoopVertexCheckers() {
+    final AtomicInteger distinctIntegerForEmptyOutputTag = new AtomicInteger(0);
```

Comment:
Can you add a comment on why this is needed for loops, and why the existing private method, groupOutEdgesByAdditionalOutputTag, can't be used?

Reply:
I'll mention that the empty output tags need to be distinguished separately in loop vertices, whereas it is okay in the EncodingCompressionChecker. Regarding the AtomicInteger class, I've looked at the article that you've provided, and have encountered the following answer which takes the approach that I have taken: https://stackoverflow.com/a/34866143. WDYT?